storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -502,6 +504,7 @@ extern TYPELIB innodb_flush_method_typelib;
 #include "sslopt-vars.h"
 
 bool redo_catchup_completed = false;
+Log_format original_log_format;
```

Comment:
@altmannmarcelo I think we have another duplicate called log_detected_format. Can we reuse this?

Reply:
no orignal_log_format and log_detected_format is different. 
`
 log_format = (Log_format)mach_read_from_4(log_buf + LOG_HEADER_FORMAT);
  original_log_format = log_format;
  if (log_format == Log_format::VERSION_8_0_28 ||
      log_format == Log_format::VERSION_8_0_19) {
    // We interpret v4 & v5 as v6 by writing proper start lsn header later in
    // this function
    log_format = Log_format::VERSION_8_0_30;
  }
  log_detected_format = log_format;
  `

Reply:
We use `original_log_format` to store what was the original format.
at open tmp logfile we adjust it's header for it to be interpreted as v6 otherwise crash recovery will complain that we are trying to run a recovery (not clean shutdown) on and older version and later when we close the log file, we re-write `original_log_format` back to its header. 

We also use the original format at log_sys_init and at --prepare for deciding if we will interpret the 8th byte of each block as LOG_BLOCK_EPOCH_NO or LOG_BLOCK_CHECKPOINT_NO

Reply:
@altmannmarcelo This is wrong and its a bug. log_detected_format should be redo log version. Not the version modified for 8030 prepare

I suggest to remove one of them.

Reply:
@satya-bodapati we cannot run prepare if log_detected_format != v6 (this var comes from server code).
If we do not store the original format, we loose this information later for places where we need to switch between LOG_BLOCK_EPOCH_NO versus LOG_BLOCK_CHECKPOINT_NO.

I don't see a problem on having both.

Reply:
mach_write_compressed handles all versions. so we can store the actual version in log_detected_format and use that instead of orignal_log_format.

Reply:
@satya-bodapati  @rahulmalik87  As per our slack conversation:

log_detected_format & orignal_log_format are duplicates. We can combine them into a single var that will hold the original log version.

log_version has to be set to V6.

log_detected_format is not been set at --backup, we should set it as part of `log_file_header_deserialize` .

Will do it once we merge .29 PR here.

Reply:
Agree @altmannmarcelo